### DISCUSSION TITLE
How to make one row point to multiple rows

### DISCUSSION THREAD

**Discussion by evrazi** [supabase/supabase#8955] [Questions] [answered]
Is it possible to add a foreign key relation from one row to multiple. So one row would point to every occurrence of that id in the related table

What I mean

Contract_table
unique_id | contract_id _(this column would have a foreign key relation to Items_table.contract_id)_
0               | 1                 
1               | 2                

Items_table
unique_id | contract_id | item_name
0               | 1                 | chair         
1               | 1                 | sofa          
2               | 2                 | car            

Items with unique_id of 0 and 1 are both part of the same contract. So when that contract is retrieved it would also retrieve both items.

So there would be multiple rows shown here (this screenshot is only an example)
![supabaseExample](https://user-images.githubusercontent.com/73037066/189934334-ae464189-ef5e-4acd-9035-970ff35d42d1.png)

Watching Supabase's One to Many Relationship - SupabaseTips yt video only shows how this is possible if the foreign key relation would be on Items_table.contract_id and point towards Contract_table.contract_id so that would mean it can only work if the column has a unique value in each row which is not an option in my case.

**Comment by GaryAustin1** [comment, marked as answer]
You can have a foreign key from items_table to contracts table.  Just make the contract_id column in items_table a foreign key to contract_table.contract_id like you suggest.  The only unique requirement is where the fk points TO.  You can have several fk's pointing to the same id in another table.

**Reply by evrazi** [reply]
If I understand you correctly you are saying that I should add a foreign key on Items_table.contract_id that points towards Contract_table.contract_id? So if I retrieve any row from Contract_table would that make supabase also retrieve all rows from Items_table with the same contract_id.

Because if I understand supabase correctly it would not retrieve them. 
Doing what u suggest would retrieve a row from Contract_table whenever I retrieve any row from Items_table. 
But what I need is that when I retrieve a row from Contract_table it comes with all items that have the same contract_id.

**Reply by GaryAustin1** [reply]
Try it, it should work.  You don't have to have the fk pointing FROM the table you are selecting, it can point TO the table you are selecting and will get pulled in.

**Reply by evrazi** [reply]
My mistake, you are correct. Tested it inside an app instead of looking at it from app.supabase and it works.

Thanks.
